### ISSUE TITLE
Issue with DYLD_LIBRARY_PATH and conda on OS X

### ISSUE THREAD

**Issue by snorfalorpagus** [pywr/pywr#132]
When GLPK and Pywr are installed using Anaconda the system fails to find the relevant .dylib shared libraries at runtime. Adding the conda lib folder to the `DYLIB_LIBRARY_PATH` variable manually is a work-around for this. Need to find how to bake-in the location - using rpath?

```
export DYLD_LIBRARY_PATH=~/miniconda3/envs/pywr/lib
```

**Comment by snorfalorpagus** [comment]
Relevant issue:

https://github.com/conda/conda-build/issues/279

When running in develop mode the link can be fixed manually with `install_name_tool`:

```
install_name_tool -change "@rpath/libglpk.40.dylib" "/Users/snorf/miniconda3/envs/pywr/lib/libglpk.40.dylib" cython_glpk.so
```

**Comment by snorfalorpagus** [comment]
Better yet the rpath can be set during compilation:

```
export C_INCLUDE_PATH=${PREFIX}/include
export CFLAGS="-Wl,-rpath,${PREFIX}/lib"
$PYTHON setup.py build_ext --with-glpk --with-lpsolve install
```

I've got a conda build working with GLPK and lpsolve on OS X now.

```
===== testing package: pywr-0.1-np110py34_0 =====
running run_test.py
============================= test session starts ==============================
platform darwin -- Python 3.4.4, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
solver: glpk
rootdir: /Users/snorf/miniconda3/conda-bld/work, inifile: 
collected 90 items 

../work/tests/test_analytical.py .......x...
../work/tests/test_control_curves.py ..
../work/tests/test_core.py .....x...........
../work/tests/test_groundwater.py .
../work/tests/test_license.py ......
../work/tests/test_parameters.py ........
../work/tests/test_piecewise.py ..
../work/tests/test_recorders.py ........
../work/tests/test_river.py ....
../work/tests/test_run.py .....xx.xxx...xxx...x.x...
../work/tests/test_scenarios.py .....

==================== 78 passed, 12 xfailed in 3.78 seconds =====================
============================= test session starts ==============================
platform darwin -- Python 3.4.4, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
solver: lpsolve
rootdir: /Users/snorf/miniconda3/conda-bld/work, inifile: 
collected 90 items 

../work/tests/test_analytical.py .......x...
../work/tests/test_control_curves.py ..
../work/tests/test_core.py .....x...........
../work/tests/test_groundwater.py .
../work/tests/test_license.py ......
../work/tests/test_parameters.py ........
../work/tests/test_piecewise.py ..
../work/tests/test_recorders.py ........
../work/tests/test_river.py ....
../work/tests/test_run.py .....xx.xxx...xxx...x.x...
../work/tests/test_scenarios.py .....

==================== 78 passed, 12 xfailed in 1.84 seconds =====================
===== pywr-0.1-np110py34_0 OK =====
TEST END: pywr-0.1-np110py34_0
# If you want to upload this package to anaconda.org later, type:
#
# $ anaconda upload /Users/snorf/miniconda3/conda-bld/osx-64/pywr-0.1-np110py34_0.tar.bz2
#
# To have conda build upload to anaconda.org automatically, use
# $ conda config --set anaconda_upload yes
```

**Comment by snorfalorpagus** [comment]
Merged as a754f87756d533b487cc4d06ee8c6c0f25e107ca.

Need to write up a blog post on this.
